draft-mouris-cfrg-mastic.md

```diff
@@ -205,11 +205,42 @@ into Mastic in {{vidpf}}. The appendix includes supplementary material:
+* `flp.decode(output: list[F], num_measurements: int) -> AggResult` maps a sum of aggregate shares to an aggregate result.
+* `flp.encode(measurement: Measurement) -> list[F]` encodes a raw measurement as a vector of field elements. The return value MUST be of length `MEAS_LEN`.
+* `flp.prove(meas: list[F], prove_rand: list[F], joint_rand: list[F]) -> list[F]` is a prover's deterministic proof-generation algorithm. Its inputs are the encoded measurement, the "prover randomness" `prove_rand`, and the "joint randomness" `joint_rand`. The prover randomness is used only by the prover, but the joint randomness is shared by both the prover and verifier.
+* `flp.query(meas: list[F], proof: list[F], query_rand: list[F], joint_rand: list[F], num_shares: int) -> list[F]` is a verifier's query algorithm. This is used to "query" the measurement and proof. The result of the query (i.e., the output of this function) is called the "verifier message". In addition to the measurement and proof, this algorithm takes as input the query randomness `query_rand` and the joint randomness `joint_rand`. The former is used only by the verifier. Here, num_shares specifies how many shares were generated.
+* `flp.truncate(meas: list[F]) -> list[F]` maps an encoded measurement (e.g., the bit-encoding of the measurement) to an aggregatable output (e.g., the singleton vector containing the measurement). The length of the input MUST be `MEAS_LEN` and the length of the output MUST be `OUTPUT_LEN`.
+* `front(length: int, vec: list[Any]) -> (list[Any], list[Any])` splits `vec` into two vectors, where the first vector is made up of the first length elements of the input, i.e., `(vec[:length], vec[length:])`.
+* `gen_rand(len: int) -> bytes` returns an array of random bytes generated by a cryptographically secure pseudorandom number generator (CSPRNG). The length of output MUST be `len`.
+* `len(obj: Sized) -> int` returns the number of items in `obj`. The object argument can be a sequence (e.g., string, list, or tuple) or a collection (e.g., dictionary, set).
+* `list.append(elem: Any) -> None` adds a single element `elem` to the end of a `list` instance.
+* `list.copy() -> list` returns a shallow copy of a `list` instance.
+* `range(stop)` or `range(start, stop[, step])` is the range function from the Python standard library. The one-argument form returns the integers from zero (inclusive) to `stop`, exclusive. The two- and three-argument forms allow overriding the start of the range and overriding the step between successive output values.
+* `set(iterable: Optional[Iterable] = None) -> set` creates a new set object, which is an unordered collection of unique elements. The optional `iterable` argument (e.g., list, tuple, or string) is used to initialize the set. If no argument is provided, an empty set is created.
```

Comment:
I don't think we use this in the Mastic spec, but worth double checking.

Reply:
!VDAF also lists basic functions (like range)

Reply:
My point is that we don't need to define it  unless we actually use it. Do we?

Reply:
Sure we do (using 8fa0a9c):
append: L458
copy: 382
range: L385
set: L500
!VDAF also defines such functions: https://github.com/cfrg/draft-irtf-cfrg-vdaf/blob/bd983637a87a7fa8353db5771c184585b2550271/draft-irtf-cfrg-vdaf.md?plain=1#L858

Reply:
Ack, thanks for verifying.